deno_dist/jsx/dom/index.ts

```diff
@@ -36,3 +53,55 @@ export const cloneElement = <T extends JSXNode | JSX.Element>(
     (element as JSXNode).key
   ) as T
 }
+
+export {
+  useState,
+  useEffect,
+  useRef,
+  useCallback,
+  use,
+  startTransition,
+  useTransition,
+  useDeferredValue,
+  startViewTransition,
+  useViewTransition,
+  useMemo,
+  useLayoutEffect,
+  useReducer,
+  useDebugValue,
+  Suspense,
+  ErrorBoundary,
+  createContext,
+  useContext,
+  memo,
+  isValidElement,
+  createElement,
+  cloneElement,
+}
+
+export default {
```

Comment:
Does not it export `jsx`? I know it's not necessary only to make it compatible with React. So, it might not be the problem, but I'm just curious.

Reply:
@yusukebe Thank you!
I've been struggling with this.

I think `jsx` will be used in the old React import style, as shown below. I didn't add it because I didn't think it would be used in "hono/jsx/dom" usage.  I think "hono/jsx/dom" is only used with "jsxImportSource".

```ts
/* @jsx jsx */
import { jsx } from "hono/jsx/dom";
```

That said, `jsx` has been exported from 'hono/jsx' and I think you are right that it would be easier to understand if `jsx` was exported from “hono/jsx/dom" as well, as you pointed out.

What should we do, export it?

Reply:
@usualoma 

I was wondering, but I think it should export `jsx`. Not many people may use `jsx`, but I may sometimes use `jsx` directly.

For example, in the case of HonoX, `jsx` is imported [here](https://github.com/honojs/honox/blob/a186c42bfd35c05954df614078e89fcb19c1263c/src/client/client.ts#L2C1-L2C56) from `hono/jsx/dom/jsx-runtime`, but `hono/jsx/dom` is better.

Reply:
@usualoma 

Ah, perhaps, if that HonoX case, can we use `createElement`?

Reply:
@yusukebe Oh, I see.

It is customary for a function named `jsx` to be exported from '*/jsx-runtime’. I think that [this](https://github.com/honojs/honox/blob/a186c42bfd35c05954df614078e89fcb19c1263c/src/client/client.ts#L2C1-L2C56) is acceptable to use. 

```ts
import { jsx } from "react/jsx-runtime";
console.log(jsx('div', {}, 'Hello, World!'))
```

```ts
import { jsx } from "preact/jsx-runtime";
console.log(jsx('div', {}, 'Hello, World!'))
```


Also, since `jsx` is an optimized version of `createElement`, I think it is not a bad idea to use `import { jsx as jsxFn } from '*/jsx-runtime'` instead of `createElement`.


### About HonoX

I don't think the current code that I am importing from hono/jsx/dom/jsx-runtime will result in minimal output and I don't think it is bad usage. But on the other hand,  you are right, `import { createElement } from 'hono/jsx/dom'` also makes sense since it is intended to be used as a `createElement`.

I see no problem either way.

Reply:
@usualoma 

Thanks for the explanation! Then, how about go with "exporting `jsx`"? One more reason is that `hono/jsx` is exporting `jsx`, so I want to make it the same as that.

If you agree, could you add that?

Reply:
@yusukebe OK
Exported by b188977cdfa5ac0421e03a4fd9f7103978f6d85d!

Reply:
@usualoma 

Thanks! Looks good! I'll merge this later.